src/object/srv_obj_migrate.c

```diff
@@ -1762,82 +1761,54 @@ migrate_insert_recxs_sgl(daos_iod_t *iods, daos_epoch_t **iods_ephs, uint32_t *i
 			break;
 	}
 
-	/* None duplicate iods, let's create new one */
-	if (i == *iods_num) {
-		rc = daos_iod_copy(&iods[i], new_iod);
+	/* This IOD already exists, let's check if iod_size and type matched */
+	if (iods[i].iod_type != DAOS_IOD_NONE &&
+	    (iods[i].iod_size != new_iod->iod_size ||
+	     iods[i].iod_type != new_iod->iod_type ||
+	     iods[i].iod_type == DAOS_IOD_SINGLE)) {
```

Comment:
because SINGLE value should not have the same name at all.  

Since this is from the remote fetching recxs, so probably using return -EINVAL, instead of assertion?